std/experimental/allocator/building_blocks/free_list.d

```diff
@@ -3,6 +3,24 @@ module std.experimental.allocator.building_blocks.free_list;
 import std.experimental.allocator.common;
 import std.typecons : Flag, Yes, No;
 
+// indicates wether A is an allocator suitable to be used for a FreeList.
+private template isFreeListAlloc(A)
+{
+    enum isFreeListAlloc =
+        (is(A == class) || is(A == struct) || is(A == interface)) &&
+        __traits(hasMember, A, "allocate") &&
+        (__traits(hasMember, A, "deallocate") ||
+        __traits(hasMember, A, "deallocateAll"));
+}
```

Comment:
No. That's not what we had in mind. None of the allocators should have constraints like this. 
IAllocator has more than 10 members. If you took that road, you would need to make more than 2^10 constraints for every possible subset of methods which is clearly wrong.

Instead you should use the MDFINE pattern:
* The allocator *types* should not have any constraints, maybe except for `allocate`, which is (AFAIK) the only hard requirement. But since types can't be overloaded it should be a static assert and not a template constraint, because that way we can provide a better message. AFAIR most allocators assume that the parent allocator has `allocate` and that's fine becuase the compiler would complain otherwise.
* Every individual *method* should be *conditionally defined* (via static if (hasMethod...), and not restricted with a constraint), depending on whether the parent allocator can provide support for it.

---

MDFINE = Method Definition Failure Is Not an Error.

You can find more from Andrei's talks at CppCon and DConf 2015